sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Helpers/SymbolHelper.cs

```diff
@@ -82,6 +82,25 @@ public static bool TryGetOverriddenOrInterfaceMember<T>(this T symbol, out T ove
                 return overriddenSymbol != null;
             }
 
+            return TryGetInterfaceMember(symbol, out overriddenSymbol);
+        }
+
+        public static bool TryGetInterfaceMember<T>(this T symbol, out T overriddenSymbol)

```

Comment:
This method is very similar to TryGetOverriddenOrInterfaceMember and there is no need to check for null before calling CanSymbolBeInterfaceMemberOrOverride. Can we join them? An idea:

        public static bool TryGetInterfaceMember<T>(this T symbol, out T overriddenSymbol, bool includeOverrides)
            where T : class, ISymbol
        {
            if (!CanSymbolBeInterfaceMemberOrOverride(symbol))
            {
                if (includeOverrides && symbol.IsOverride)
                {
                    overriddenSymbol = GetOverriddenMember(symbol);
                    return overriddenSymbol != null;
                }

                overriddenSymbol = null;
                return false;
            }

            overriddenSymbol = symbol.ContainingType
                .AllInterfaces
                .SelectMany(@interface => @interface.GetMembers())
                .OfType<T>()
                .FirstOrDefault(member => symbol.Equals(symbol.ContainingType.FindImplementationForInterfaceMember(member)));
            return overriddenSymbol != null;
        }


        private static bool CanSymbolBeInterfaceMemberOrOverride(ISymbol symbol)
        {
            return symbol is IMethodSymbol ||
                symbol is IPropertySymbol ||
                symbol is IEventSymbol;
        }